public/partials/farm_new.html

```diff
@@ -50,7 +51,9 @@
     </md-input-container>
 
     <md-input-container layout="row" layout-align="center center">
-        <md-button ng-click="cancelAdd()">Cancel</md-button>
-        <md-button ng-click="addNewFarm()">Save</md-button>
+        <md-button ng-if="newFarm" ng-click="cancelAdd()">Cancel</md-button>
```

Comment:
@nick22891 The showing (and hiding) of buttons according to the state of the form could be handled better. I would suggest viewing code done in implementing the admin dashboard and refactor this code.